lib/dor/text_extraction/abbyy/split_alto.rb

```diff
@@ -24,31 +27,30 @@ def output_path(filename)
           File.join(File.dirname(alto_path), filename)
         end
 
-        # rubocop:disable Metrics/AbcSize
         def xml_structure(page)
-          namespaces = { 'xsi:schemaLocation': 'http://www.loc.gov/standards/alto/ns-v3# http://www.loc.gov/alto/v3/alto-3-1.xsd' }.merge(doc.collect_namespaces)
           Nokogiri::XML::Builder.new do |xml|
             xml.alto(namespaces) do
-              doc.css('//OCRProcessing').each { |node| xml.parent << node.dup }
-              doc.css('//Styles').each { |node| xml.parent << node.dup }
+              xml.parent << ocr_processing_nodes
+              xml.parent << style_nodes
               xml.parent << page
             end
           end
         end
 
-        def page_filenames
-          @page_filenames ||= doc.css('//sourceImageInformation').children.select(&:element?).map(&:text)
+        def namespaces
+          @namespaces ||= { 'xsi:schemaLocation': 'http://www.loc.gov/standards/alto/ns-v3# http://www.loc.gov/alto/v3/alto-3-1.xsd' }.merge(doc.collect_namespaces)
         end
-        # rubocop:enable Metrics/AbcSize
 
-        def split_to_files
-          pages = doc.css('//Page')
-          {}.tap do |file_hash|
-            pages.each_with_index do |page, i|
-              filename = "#{File.basename(page_filenames[i], File.extname(page_filenames[i]))}.xml"
-              file_hash[filename] = xml_structure(page).to_xml
-            end
-          end
```

Comment:
this method is gone - we aren't going to split in memory into a hash anymore since we don't need to ... we will just iterate over `pages` and write to disk immediately